shlangus/node-cloud-be
# Task 9

## Scope
I hope I've completed all the required and additional tasks.

## Links
Product Service: https://ebygl9njp0.execute-api.eu-west-1.amazonaws.com/dev/products
Cart Service: http://shlangus-cart-api-dev.eu-west-1.elasticbeanstalk.com/api/profile/cart
BFF Service: http://shlangus-bff-service-dev.eu-west-1.elasticbeanstalk.com/products

## How to test
Click links below and see that responses are same :)

GET /products
Direct: https://ebygl9njp0.execute-api.eu-west-1.amazonaws.com/dev/products
BFF: http://shlangus-bff-service-dev.eu-west-1.elasticbeanstalk.com/products

GET /products/{id} 
Direct: https://ebygl9njp0.execute-api.eu-west-1.amazonaws.com/dev/products/712ae831-65ff-44cc-9c1a-a7eea6feb676
BFF: http://shlangus-bff-service-dev.eu-west-1.elasticbeanstalk.com/products/712ae831-65ff-44cc-9c1a-a7eea6feb676

An error also should be the same
Direct: https://ebygl9njp0.execute-api.eu-west-1.amazonaws.com/dev/products/712ae831-65ff-44cc-9c1a-a7eea6feb600
BFF: http://shlangus-bff-service-dev.eu-west-1.elasticbeanstalk.com/products/712ae831-65ff-44cc-9c1a-a7eea6feb600

GET /cart
Direct: http://shlangus-cart-api-dev.eu-west-1.elasticbeanstalk.com/api/profile/cart
BFF: http://shlangus-bff-service-dev.eu-west-1.elasticbeanstalk.com/cart

The following requests are in WebStorm Http Client format, modify them as needed:
```
PUT http://shlangus-cart-api-dev.eu-west-1.elasticbeanstalk.com/api/profile/cart
#PUT http://shlangus-bff-service-dev.eu-west-1.elasticbeanstalk.com/cart
Content-Type: application/json

{
  "items": [
    {
      "product": {
        "id": "712ae831-65ff-44cc-9c1a-a7eea6feb676",
        "title": "Tomato",
        "description": "Red, not a fruit, so not sure why we sell it",
        "price": "5",
        "count": 1
      },
      "count": 2
    }
  ]
}

POST https://ebygl9njp0.execute-api.eu-west-1.amazonaws.com/dev/products
#POST http://shlangus-bff-service-dev.eu-west-1.elasticbeanstalk.com/products
Content-Type: application/json

{
  "title":"Tomato7000","description":"Test Tomato","price":5,"count":1
}
```
